edxval/api.py

```diff
@@ -733,6 +733,30 @@ def get_videos_for_course(course_id, sort_field=None, sort_dir=SortDirection.asc
     )
 
 
+@cachetools.func.ttl_cache(maxsize=None, ttl=settings.TTL_CACHE_TIMEOUT)
+def get_transcript_languages(course_id, provider_type):
+    """
+    Returns a list of languages for which transcripts are available for a course
+
+    Args:
+        course_id (str): course id
+        provider_type (str): transcript provider type
+
+    Returns:
+        (list): A list of language codes
+    """
+    transcript_languages = set()
+    video_transcripts = VideoTranscript.objects.filter(
+        video__edx_video_id__in=CourseVideo.objects.filter(
+            course_id=course_id
+        ).values_list("video__edx_video_id", flat=True)
+    ).select_related("video",)
```

Comment:
What is the purpose of this select related?

Reply:
Why do we need the `edx_video_id` field? Can we not just query like 
`VideoTranscript.objects.filter(video_id__in=CourseVideo.objects.filter(course_id=course_id).values_list('video_id'))`

Reply:
`api.py` already has various places where it's currently using `edx_video_id`, so I went with it . Didn't know that we have `video_id` in the schema.

Refs:
https://github.com/openedx/edx-val/blob/b7c5db6c95c9a8655e1f3d58429c2e52b4ead66e/edxval/api.py#L814
https://github.com/openedx/edx-val/blob/b7c5db6c95c9a8655e1f3d58429c2e52b4ead66e/edxval/api.py#L838